docs/draft/17-url-data.md

```diff
@@ -0,0 +1,118 @@
+---
+permalink: /draft/17
+title: 17/URL data
+parent: Draft specs
+layout: default
+---
+
+# 17/URL data
+
+> Version: 0.1
+>
+> Status: Draft
+>
+> Authors: Felicio Mununga <felicio@status.im>
+
+## Abstract
+
+This document describes serialization, compression and encoding of data within URLs, so the app links could carry enough information to help users with instant previewing and verifying of their contents even prior clicking or taking and action on the content.
+
+## Related scope
+
+### Specs
+
+- URL scheme
+
+### Features
+
+- Onboarding website
+- Link preview
+- Link sharing
+- Deep linking
+- Routing and navigation
+
+### App entities
+
+- Community
+- Channel
+- User
+
+## Requirements
+
+### Product
+
+- Encode data in URL until Waku response is instant and considered reliable enough for this use case, or until a product roadmap for alternative decentrilzed storage is finalized
+- Verify state while loading website until encoded data is matched against Waku response to mitigate identity attacks
+- Accept only `[A-Za-z0-9_-.\u0020]` character class for textual fields
+- Shortest result possible
+
+## Implementation
+
+## Example
+
+- See <https://github.com/felicio/status-web/blob/34535355d46593179c997ec5ef0eda32547890aa/packages/status-js/src/utils/encode-url-data.test.ts>
+
+### Data
+
+#### Fields
+
+```protobuf
+syntax = "proto3";
+
+message CommunityPreview {
+ string display_name = 3;
+ string description = 4;
+ uint32 members_count = 5;
+ string color = 8;
+}
+
+message ChannelPreview {
+ string display_name = 2;
+ string description = 3;
+ string emoji = 5;
+ string color = 6;
+ CommunityPreview community = 7;
+}
+
+message UserPreview {
+ string display_name = 2;
+ string description = 3;
+ string color = 7;
+}
+
+message URLData {
+ // Community, Channel, or User
+ bytes content = 1;
+ // content and public key concatanated and sha256 hashed twice, returning first 4 bytes
+ bytes checksum = 2;
+}
+
+```
```

Comment:
Since this spec is written from scratch ~ would it be possible to store this metadata on ipfs?

```suggestion
#### Fields

```protobuf
syntax = "proto3";

message CommunityPreview {
 bytes community_id = 1;
 bytes ipfs_cid = 2;
}

message ChannelPreview {
 bytes channel_id = 1;
 bytes community_id = 2;
 bytes ipfs_cid = 3;
}

message UserPreview {
 string display_name = 2;
 string description = 3;
 string color = 7;
}

message URLData {
 // Community, Channel, or User
 bytes content = 1;
 // content and public key concatanated and sha256 hashed twice, returning first 4 bytes
 bytes checksum = 2;
}
```

cc: @alrevuelta re dos prevention

this shouldn't be difficult for mobile devices to do, esp. since they don't need waku to find this data.

Reply:
I raised that in https://github.com/status-im/status-web/issues/327#issuecomment-1422764275, however product decided not proceed with IPFS now.

c/o @John-44

Reply:
@alrevuelta it's not possible to use an IPFS pinning service that's either run by Status or paid for by Status for legal reasons.  Without an IPFS pinning service it would not be sufficiently reliable.

Post MVP we will probably build a decenteralized solution for IPFS pinning that doesn't involve Status/Logos the org in any way, but we can't create a dependency on future infrastructure that doesn't yet exist (and won't exist until post-MVP) for a must-have MVP item (the web preview screens are must-have for MVP)